src/table/virtualized-table/hooks/use-scrollbar-width.ts

```diff
@@ -0,0 +1,26 @@
+import { useEffect, useState } from 'react';
+import useOnPropsChange from './use-on-props-change';
+
+const useScrollbarWidth = (ref: React.RefObject<HTMLDivElement>) => {
+  const [xScrollbarWidth, setXScrollbarWidth] = useState(0);
+  const [yScrollbarWidth, setYScrollbarWidth] = useState(0);
+  const { offsetWidth = 0, offsetHeight = 0, clientWidth = 0, clientHeight = 0 } = ref.current ?? {};
+  const widthDiff = offsetWidth - clientWidth;
+  const heightDiff = offsetHeight - clientHeight;
+
+  useEffect(() => {
+    if (ref.current) {
+      setYScrollbarWidth(ref.current.offsetWidth - ref.current.clientWidth);
+      setXScrollbarWidth(ref.current.offsetHeight - ref.current.clientHeight);
+    }
+  }, [ref]);
+
+  useOnPropsChange(() => {
+    setYScrollbarWidth(widthDiff);
+    setXScrollbarWidth(heightDiff);
+  }, [widthDiff, heightDiff]);
```

Comment:
I missed when this hook was introduced, had a look at  the definition. How is it different from a useEffect?

Reply:
It's actually a wrapper for this https://beta.reactjs.org/learn/you-might-not-need-an-effect#adjusting-some-state-when-a-prop-changes and turns out be useful in this case as well.